src/main/java/com/linkedin/kmf/services/MultiClusterTopicManagementService.java

```diff
@@ -0,0 +1,343 @@
+      _executor.scheduleWithFixedDelay(r, 0, _scheduleIntervalMs, TimeUnit.MILLISECONDS);
+      LOG.info("{}/MultiClusterTopicManagementService started.", _serviceName);
+    }
+  }
+
+  @Override
+  public synchronized void stop() {
+    if (_isRunning.compareAndSet(true, false)) {
+      _executor.shutdown();
+      LOG.info("{}/MultiClusterTopicManagementService stopped.", _serviceName);
+    }
+  }
+
+  @Override
+  public boolean isRunning() {
+    return _isRunning.get() && !_executor.isShutdown();
+  }
+
+  @Override
+  public void awaitShutdown() {
+    try {
+      _executor.awaitTermination(Integer.MAX_VALUE, TimeUnit.MILLISECONDS);
+    } catch (InterruptedException e) {
+      LOG.info("Thread interrupted when waiting for {}/MultiClusterTopicManagementService to shutdown", _serviceName);
+    }
+    LOG.info("{}/MultiClusterTopicManagementService shutdown completed", _serviceName);
+  }
+
+  private class TopicManagementRunnable implements Runnable {
+    @Override
+    public void run() {
+      try {
+        for (Map props : _serviceProps.values()) {
+          maybeCreateTopic(props);
+        }
+
+        maybeAddPartitions(_serviceProps.values());
+
+        for (Map.Entry<String, Map> entry : _serviceProps.entrySet()) {
+          String clusterName = entry.getKey();
+          Map props = entry.getValue();
+          try {
+            maybeReassignPartitionAndElectLeader(props);
+          } catch (IOException | ZkNodeExistsException | AdminOperationException e) {
+            LOG.warn(_serviceName + "/MultiClusterTopicManagementService will retry later in cluster " + clusterName, e);
+          }
+        }
+      } catch (Exception e) {
+        LOG.error(_serviceName + "/MultiClusterTopicManagementService will stop due to error.", e);
+        stop();
+      }
+    }
+  }
+
+  private static void maybeCreateTopic(Map props) throws Exception {
+    TopicManagementServiceConfig config = new TopicManagementServiceConfig(props);
+    String topic = config.getString(CommonServiceConfig.TOPIC_CONFIG);
+    String zkConnect = config.getString(CommonServiceConfig.ZOOKEEPER_CONNECT_CONFIG);
+    Boolean topicCreationEnabled = config.getBoolean(TopicManagementServiceConfig.TOPIC_CREATION_ENABLED_CONFIG);
+    int replicationFactor = config.getInt(TopicManagementServiceConfig.TOPIC_REPLICATION_FACTOR_CONFIG);
+    double partitionsToBrokerRatio = config.getDouble(TopicManagementServiceConfig.PARTITIONS_TO_BROKERS_RATIO_CONFIG);
+    String topicFactoryClassName = config.getString(TopicManagementServiceConfig.TOPIC_FACTORY_CLASS_CONFIG);
+    Map topicFactoryConfig = props.containsKey(TopicManagementServiceConfig.TOPIC_FACTORY_PROPS_CONFIG) ?
+        (Map) props.get(TopicManagementServiceConfig.TOPIC_FACTORY_PROPS_CONFIG) : new HashMap();
+    TopicFactory topicFactory = (TopicFactory) Class.forName(topicFactoryClassName).getConstructor(Map.class).newInstance(topicFactoryConfig);
+
+    if (topicCreationEnabled) {
```

Comment:
If this check happens towards the beginning of the method then the user is spared from needing to have valid values for the other configuration items.